### PR TITLE
Added tabs

### PR DESCRIPTION
**Pull request by Jorolf**
This pull request is not finished.
It is for reviewing so I can change things now and not later when it's too complicated.
Please tell me if I'm doing something horribly wrong

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/TabContainer.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                    Origin = Anchor.Centre,
+                    Text = tabName,
+                    Colour = tab.TabIndexTextColor,
+                }
+            };
+
+        }
+
+
+        protected override bool OnClick(InputState state)
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: You might want to derive from `ClickableContainer`

File: osu.Framework/Graphics/UserInterface/TabContainer.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+namespace osu.Framework.Graphics.UserInterface
+{
+    public class TabContainer : Container
+    {
+
+        private Container currentTab;
+        private FlowContainer tabIndexesFlow;
+        private Container tabIndexes;
+        private Container tabBodies;
+        public float TabIndexHeight { get { return tabIndexes.Height; } set { tabIndexes.Height = value; tabBodies.Margin = new MarginPadding() { Top = value }; } }
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: Format this >_<

File: osu.Framework/Graphics/UserInterface/TabContainer.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        public Color4 TabIndexTextColor { get; set; } = Color4.Yellow;
+        public Color4 TabIndexBackgroundColor { get { return tabIndexesBackground.Colour; } set { tabIndexesBackground.Colour = value; } }
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: Format this >_<

File: osu.Framework/Graphics/UserInterface/TabContainer.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using osu.Framework.Input;
+
+namespace osu.Framework.Graphics.UserInterface
+{
+    public class TabContainer : Container
+    {
+
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: Opening curly brace should not be followed by blank line

File: osu.Framework/Graphics/UserInterface/TabContainer.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        private Container currentTab;
+        private FlowContainer tabIndexesFlow;
+        private Container tabIndexes;
+        private Container tabBodies;
```
Comment (UselessToucan) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: I would like to have 1 blank line between these private/public members

File: osu.Framework.VisualTests/Tests/TestCaseTabs.cs [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace osu.Framework.VisualTests.Tests
+{
+    class TestCaseTabs : TestCase
+    {
+
```
Comment (UselessToucan) [inline osu.Framework.VisualTests/Tests/TestCaseTabs.cs]: Opening curly brace should not be followed by blank line

File: osu.Framework/Graphics/UserInterface/TabContainer.cs (hunk 1 above)
Comment (Jorolf) [inline osu.Framework/Graphics/UserInterface/TabContainer.cs]: What's the difference?

**Comment by peppy** [comment]
closing until ready for review (also fix CO failing)

**Comment by peppy** [comment]
*CI

**Pull request closed without merging**